core/src/main/java/org/apache/accumulo/core/client/PluginEnvironment.java

```diff
@@ -158,6 +167,18 @@
    */
   <T> T instantiate(String className, Class<T> base) throws Exception;
 
+  /**
+   * Loads a class using Accumulo's system classloader.
+   *
+   * @param className
+   *          Fully qualified name of the class.
+   * @param base
+   *          The expected super type of the class.
+   *
+   * @since 2.1.0
+   */
+  <T> Class<? extends T> loadClass(String className, Class<T> base) throws ClassNotFoundException;
```

Comment:
What's the use case for this being in the public API?
Note: we're transitioning how we do class loading, and phasing out some internal class loading stuffs. We should avoid exposing stuff like this to the public API. The user should be able to just load the class themselves, using their own object's classloader, if they need to.

Reply:
Fair enough. I really just needed the context-loading variant for TableLoadBalancer. But that really just needs a method to get the context name for the table configuration, so I added that method to BalancerEnvironment and removed the loadClass methods from here.